scalyr_agent/builtin_monitors/journald_monitor.py

```diff
@@ -435,3 +559,169 @@ def gather_sample(self):
+        log_config["path"] = full_path
+
+    return log_config
+
+
+class LogManager:
+    def __init__(
+        self,
+        default_config,
+        module_name,
+        global_config,
+        file_name_template,
+        max_log_size=20 * 1024 * 1024,
+        max_log_rotations=2,
+    ):
+        self._file_template = file_name_template
+        self.loggers = {}
+        self.default_config = default_config
+        self._max_log_size = max_log_size
+        self._max_log_rotations = max_log_rotations
+        self.module_name = module_name
+        self.global_config = global_config
+        self.log_watcher = None
+        self.all_configs = self.global_config.get_all_log_configs([])
+        self.loggers_created = False
+
+        self.initialize()
+
+    def initialize(self):
+        """Called at the end of __init__, intended for overriding by subclasses
+        If configuration can be invalid, it should be checked here
+        """
+        for config in self.all_configs:
+            match = self.match_config_to_logger(config)
+            if match:
+                regex = re.compile(match)
+                if match not in self.loggers:
+                    self.loggers[match] = {}
+                self.loggers[match]["regex"] = regex
+
+    def match_line_to_logger(self, data, extra_fields=None):
+        """Get the unique identifier of the logger we should output this data to, intended for overriding by subclasses
+
+        @param data: Data to be send to the logger, used for matching
+        @param extra_fields: Any extra data used for matching
+        @return: A string uniquely identifying the matched logger
+        """
+        if extra_fields and "unit" in extra_fields:
+            for match in self.loggers.keys():
+                if match != ".*" and self.loggers[match]["regex"].match(
+                    extra_fields["unit"]
+                ):
+                    return match
+        return None
+
+    def match_config_to_logger(self, config):
+        """Get the unique identifier of the logger that should have this config, intended for overriding by subclasses
+
+        @param config: The configuration for a logger
+        @return: A string uniquely identifying the matched logger
+        """
+        if "journald_unit" in config:
+            return config["journald_unit"]
+        return None
+
+    def set_log_watcher(self, log_watcher):
+        """Set the log_watcher so we can use it to register new log files
+        If loggers have not yet been created they will be when this is called
+
+        @param log_watcher:
+        @return:
+        """
+        self.log_watcher = log_watcher
+        if not self.loggers_created:
+            for config in self.all_configs:
+                match = self.match_config_to_logger(config)
+                if match:
+                    self.create_logger(match, config)
+            self.loggers_created = True
+
+    def emit(self, data, extra_fields=None):
```

Comment:
Remove this `emit` method.  We will now just rely on the user of this abstraction to call `info` on the logger returned from `get_logger`.